src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -92,17 +93,23 @@ public ReadsSparkSource(final JavaSparkContext ctx, final ValidationStringency v
 
         setHadoopBAMConfigurationProperties(readFileName, referencePath);
 
+        boolean isBam = IOUtils.isBamFileName(readFileName);
+        if (isBam && intervals != null && !intervals.isEmpty()) {
+            BAMInputFormat.setIntervals(conf, intervals);
```

Comment:
Have we tested `BAMInputFormat.setIntervals()` with a large interval list like `ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/2.8/b37/Broad.human.exome.b37.interval_list.gz`? I wonder whether there might be performance issues, given that it converts all the intervals into one huge String and re-parses them later.